src/llmq/quorums_signing_shares.cpp

```diff
@@ -229,34 +229,79 @@ void CSigSharesManager::ProcessMessage(CNode* pfrom, const std::string& strComma
+                return;
+            }
+        }
+        // not accounting as pending msg (as it's not batched in any way)
     } else if (strCommand == NetMsgType::QSIGSHARESINV) {
-        CSigSharesInv inv;
-        vRecv >> inv;
-        ProcessMessageSigSharesInv(pfrom, inv, connman);
+        std::vector<CSigSharesInv> msgs;
+        vRecv >> msgs;
+        if (msgs.size() > MAX_MSGS_CNT_QSIGSHARESINV) {
+            LogPrint("llmq", "CSigSharesManager::%s -- too many invs in QSIGSHARESINV message. cnt=%d, max=%d, node=%d\n", __func__, msgs.size(), MAX_MSGS_CNT_QSIGSHARESINV, pfrom->id);
+            BanNode(pfrom->id);
+            return;
+        }
+        for (auto& inv : msgs) {
+            if (!ProcessMessageSigSharesInv(pfrom, inv, connman)) {
+                return;
+            }
+        }
+        LOCK(cs);
+        nodeStates[pfrom->id].pendingReceivedMsgCount += msgs.size();
     } else if (strCommand == NetMsgType::QGETSIGSHARES) {
-        CSigSharesInv inv;
-        vRecv >> inv;
-        ProcessMessageGetSigShares(pfrom, inv, connman);
+        std::vector<CSigSharesInv> msgs;
+        vRecv >> msgs;
+        if (msgs.size() > MAX_MSGS_CNT_QGETSIGSHARES) {
+            LogPrint("llmq", "CSigSharesManager::%s -- too many invs in QGETSIGSHARES message. cnt=%d, max=%d, node=%d\n", __func__, msgs.size(), MAX_MSGS_CNT_QGETSIGSHARES, pfrom->id);
+            BanNode(pfrom->id);
+            return;
+        }
+        for (auto& inv : msgs) {
+            if (!ProcessMessageGetSigShares(pfrom, inv, connman)) {
+                return;
+            }
+        }
     } else if (strCommand == NetMsgType::QBSIGSHARES) {
-        CBatchedSigShares batchedSigShares;
-        vRecv >> batchedSigShares;
-        ProcessMessageBatchedSigShares(pfrom, batchedSigShares, connman);
+        std::vector<CBatchedSigShares> msgs;
+        vRecv >> msgs;
+        size_t totalSigsCount = 0;
+        for (auto& bs : msgs) {
+            totalSigsCount += bs.sigShares.size();
+        }
+        if (totalSigsCount > MAX_MSGS_TOTAL_BATCHED_SIGS) {
+            LogPrint("llmq", "CSigSharesManager::%s -- too many sigs in QBSIGSHARES message. cnt=%d, max=%d, node=%d\n", __func__, msgs.size(), MAX_MSGS_TOTAL_BATCHED_SIGS, pfrom->id);
+            BanNode(pfrom->id);
+            return;
+        }
+        for (auto& bs : msgs) {
+            if (!ProcessMessageBatchedSigShares(pfrom, bs, connman)) {
+                return;
+            }
+        }
+        LOCK(cs);
+        nodeStates[pfrom->id].pendingReceivedMsgCount += msgs.size();
     }
 }
 
-void CSigSharesManager::ProcessMessageSigSesAnn(CNode* pfrom, const CSigSesAnn& ann, CConnman& connman)
+bool CSigSharesManager::ProcessMessageSigSesAnn(CNode* pfrom, const CSigSesAnn& ann, CConnman& connman)
 {
     auto llmqType = (Consensus::LLMQType)ann.llmqType;
     if (!Params().GetConsensus().llmqs.count(llmqType)) {
         BanNode(pfrom->id);
-        return;
+        return false;
```

Comment:
Could probably also move all `BanNode`s from `ProcessMessage*` to `ProcessMessage` now (for `false` cases) to keep them all in one place.

Reply:
Done